csharp/src/Ice/Connection.cs

```diff
@@ -2184,12 +2184,11 @@ private int ParseMessage(ref MessageInfo info)
                 {
                     if (_compressionSupported)
                     {
-                        info.Data = BZip2.Uncompress(info.Data, Ice1Definitions.HeaderSize, _messageSizeMax);
+                        info.Data = BZip2.Decompress(info.Data, Ice1Definitions.HeaderSize, _messageSizeMax);
                     }
                     else
                     {
-                        string lib = AssemblyUtil.IsWindows ? "bzip2.dll" : "libbz2.so.1";
-                        throw new LoadException($"cannot uncompress compressed message: {lib} not found");
+                        throw new LoadException($"cannot uncompress compressed message: {BZip2.LibName} not found");
```

Comment:
I would change the message to just "compression not supported", we don't actually know here why it's not supported... probably because the library is not found but it could also be because of a library mismatch, etc.

Reply:
What about the exception type `LoadException` , we are not loading anything at this point

Reply:
Ok perhaps just replace "not found" with "not loaded".

Reply:
I was thinking it should throw `NotSupportedException`

Reply:
Both are fine with me as long as the message is clear, it's not supported because the bzip2 library couldn't be loaded (not because Ice for C# doesn't support compression).

Reply:
I think `LoadException` is fine here, the load error is the file was not found.

Reply:
Actually it could be a different problem than file not found, so I would replace the error message by: `bzip2 library not  loaded`.